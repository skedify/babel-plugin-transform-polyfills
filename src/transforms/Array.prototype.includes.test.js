import compile from '../../test/utils/compile'

describe('Array.prototype.includes', () => {
  it('should replace Array.prototype.includes with a polyfill', () => {
    const result = compile(
      `
      const list = [1,2,3,4,5]
      if (list.includes(3) === true) {
        console.log('success')
      } else {
        console.log('failure')
      }
    `
    )

    expect(result).toMatchSnapshot()
  })

  it('should not import polyfill more than once', () => {
    const result = compile(
      `
      const list = [1,2,3,4,5]
      if (list.includes(3)) {
        console.log(list.includes(3))
      } else {
        console.log('failure')
      }
    `
    )

    expect(result).toMatchSnapshot()
  })

  it('should work when part of a call chain', () => {
    const result = compile(
      `
      const list = [1,2,3,4,5]
      if (list.slice(2).includes(3)) {
        console.log('success')
      } else {
        console.log('failure')
      }
    `
    )

    expect(result).toMatchSnapshot()
  })
})
