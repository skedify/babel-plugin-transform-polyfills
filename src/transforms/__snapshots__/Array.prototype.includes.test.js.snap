// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Array.prototype.includes should not import polyfill more than once 1`] = `
"import _Array_prototype_includes from 'babel-plugin-transform-polyfills/polyfills/Array/prototype/includes';

const list = [1, 2, 3, 4, 5];
if (_Array_prototype_includes(list, 3)) {
  console.log(_Array_prototype_includes(list, 3));
} else {
  console.log('failure');
}"
`;

exports[`Array.prototype.includes should replace Array.prototype.includes with a polyfill 1`] = `
"import _Array_prototype_includes from 'babel-plugin-transform-polyfills/polyfills/Array/prototype/includes';

const list = [1, 2, 3, 4, 5];
if (_Array_prototype_includes(list, 3) === true) {
  console.log('success');
} else {
  console.log('failure');
}"
`;

exports[`Array.prototype.includes should work when part of a call chain 1`] = `
"import _Array_prototype_includes from 'babel-plugin-transform-polyfills/polyfills/Array/prototype/includes';

const list = [1, 2, 3, 4, 5];
if (_Array_prototype_includes(list.slice(2), 3)) {
  console.log('success');
} else {
  console.log('failure');
}"
`;
